src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -303,9 +304,13 @@ public Completable start() {
         handshakeResponseSubject = CompletableSubject.create();
         handshakeReceived = false;
         CompletableSubject tokenCompletable = CompletableSubject.create();
+        if(headers != null) {
```

Comment:
```suggestion
        if (headers != null) {
```

Reply:
Or just set the headers field unconditionally